Licensing Portal: Update billing UI to display usage for daily usage pricing

atanas-dev opened the pull request:
PT: pdpAdu-bj-p2
Card: 1201902532330136-as-1201935935452157
Requires D77767-code.

#### Changes proposed in this Pull Request

* Change the Licensing Portal Billing UI based on whether the user is being billed for daily usage of licenses.
* Change the Licensing Portal Issue License UI based on whether the user is being billed for daily usage of licenses.

#### Testing instructions

* Follow the test instructions in D77767-code.
* Visit http://jetpack.cloud.localhost:3000/partner-portal/issue-license and confirm the price interval label matches your partner type.
* Visit http://jetpack.cloud.localhost:3000/partner-portal/billing and confirm the displayed quantities and prices match your partner type price interval (day vs month).
* Adjust the logic in D77767-code so your partner account is considered to have daily pricing and repeat the previous two steps.

Billing UI:
| Monthly | Daily |
| - | - |
| ![monthly](https://user-images.githubusercontent.com/22746396/161291298-973b8ce8-dc49-4514-a01d-2f260ecd92d2.png) | ![daily](https://user-images.githubusercontent.com/22746396/161986002-4eb6131b-67f3-4bbb-b035-7190e94b011d.png) |

Note: the price values are not real and are for testing purposes only. The "Days Total" numbers represent the number of days each license for a given product has been active or will be by the end of the current month.

Issue License UI:
| Monthly | Daily |
| - | - |
| ![Screen Shot 2022-03-31 at 18 53 15](https://user-images.githubusercontent.com/22746396/161097609-9a1b35e7-7cf0-4683-b3a0-587664d5bfef.png) | ![Screen Shot 2022-03-31 at 18 52 37](https://user-images.githubusercontent.com/22746396/161097483-bcae962d-9c77-414d-a19d-37e2d7791131.png) |

Note: the price values are not real and are for testing purposes only - the interval below the amount is what changes.

atanas-dev committed:
Licensing Portal: Update billing UI to display usage for daily usage pricing

atanas-dev committed:
Licensing Portal: Respect the price_interval option when issuing licenses

atanas-dev committed:
Licensing Portal: Adjust days in total wording on the billing screen

kallehauge commented on client/jetpack-cloud/sections/partner-portal/billing-details/index.tsx:
```diff
@@ -17,9 +17,13 @@ export default function BillingDetails(): ReactElement {
+					<div>{ translate( 'Unassigned licenses' ) }</div>
+					<div>
+						{ billing.isSuccess &&
+							billing.data.priceInterval === 'day' &&
```
We rely on `billing.data.priceInterval` quite often in `<BillingDetails>`. Should we define it as a variable at the top of the function? I don't see us having more than "daily vs monthly" prices anytime in the near of far future (e.g. bi-weekly?) so a `isPriceIntervalMonthly` kinda variable should be OK

kallehauge reviewed (commented):
> Follow the test instructions in D77767-code.

The instructions here are a bit bare, but it wasn't a blocker to test the PR 😄 

The code overall looks good and tests well 👍 

---

I'm not sure if it should be part of this PR, but we previously talked about showing a "Estimated cost for the rest of the month" kinda sum when we do the switch to daily billing. Is that left out in the one on purpose?

![Screenshot 2022-04-06 at 09 47 10](https://user-images.githubusercontent.com/3846700/161924811-6ed2989b-3ccb-4c3d-82d6-fc3095a9e4b9.png)

---

Should we include a description text of what "daily" means? Context: pdpAdu-gT-p2#comment-267

I'm perfectly fine with the above questions to be addressed in separate PR's

atanas-dev commented:
@kallehauge 
> showing a "Estimated cost for the rest of the month" kinda sum

I've missed letting you know about this but the value you see is actually the estimated value for the full month even if it's daily-based pricing, which is not what I initially thought would be the case. The implementation of the usage changes we introduced for daily pricing on the back end ended up reporting the full estimated usage until the end of the month rather than up to "today".
For example, if it's the 15th of the month and you see $123 as the invoice total and you don't issue or revoke any licenses until the end of the month, $123 is exactly how much you will pay.

atanas-dev commented:
@kallehauge 
I've updated the daily screenshot and added the following footnote:
> * Estimate of the combined number of full days each license will be active for by the end of the current month, accounting for licenses that were newly issued or revoked.

atanas-dev committed:
Licensing Portal: Add a daily price calculation footnote in the Billing section

kallehauge reviewed (approved):
I have one previous comment that has not been marked as "resolved", so I'm not sure if you've seen it @atanas-dev: https://github.com/Automattic/wp-calypso/pull/62350/files#r843601922

But the code itself - outside of my subjective comment - looks good and matches the findings of the P2 👍

atanas-dev committed:
Licensing Portal: Simplify billing ui rendering logic

atanas-dev commented:
Addressed the last remaining feedback item.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7222193

Thank you @atanas-dev for including a screenshot in the description! This is really helpful for our translators.

kallehauge commented on client/jetpack-cloud/sections/partner-portal/billing-details/index.tsx:
```diff
@@ -11,15 +11,16 @@ export default function BillingDetails(): ReactElement {
 	const translate = useTranslate();
 	const moment = useLocalizedMoment();
 	const billing = useBillingDashboardQuery();
+	const useDailyPrices = billing.isSuccess && billing.data.priceInterval === 'day';
```
@atanas-dev small note after the refactor that doesn't make any real difference - but if you (or I) touch the billing details list in the future, then we can remove the two instances that uses `billing.isSuccess && useDailyPrices ...` as a condition since `useDailyPrices` already contains `billing.isSuccess` 😄

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7222193).
